### DISCUSSION TITLE
dark mode

### DISCUSSION THREAD

**Discussion by kashyap-scs** [community/community#132240] [npm]
please provide dark mode for better user experiance.

**Comment by ghost** [comment]
Hi, go to settings, then to appearance and there you can set dark mode

**Reply by jmussman** [reply]
Yeah, but #gh-dark-mode-only and #light-mode-only are still not supported so if you switch to dark mode you loose images in a lot of README.md files. Kinda sucks. You would think it would only take about five lines of code to make that work...

**Reply by dinosy817** [reply]
> Yeah, but #gh-dark-mode-only and #light-mode-only are still not supported so if you switch to dark mode you loose images in a lot of README.md files. Kinda sucks. You would think it would only take about five lines of code to make that work...

Here are a few workarounds to improve the experience:

Use Alternative Image Hosts: If you can, host images on platforms that support dark mode, or provide both light and dark versions of your images.
CSS for Dark Mode: If you have control over the CSS (like in a GitHub Pages site), you could use media queries to adjust images based on the theme.
Image Formats: Use image formats that adapt well to different backgrounds (like SVGs with transparent backgrounds).
Markdown Tricks: You can add a note in your README.md to inform users about the issue and suggest switching themes as needed.

**Reply by jmussman** [reply]
The images are linked from inside the GitHub repo in the README.md using media queries to adjust for the theme. It doesn't work when you load the repo as a published package at npmjs.com. If I'm doing something wrong I'd love to learn and correct it! https://github.com/jmussman/vitest-mock-commonjs -> https://www.npmjs.com/package/vitest-mock-commonjs.

As for images that adapt well... The best you could do is an outer glow, but that really doesn't work out that well. It is what I tried for years before media queries came along :)

I think asking users to change their preferences to adapt to your limitations is not really an appropriate solution. GitHub and npmjs.com are both real. Both owned by Microsoft. How hard would it be for the media queries to work the same?

**Comment by adnuhan** [comment]
To enable Dark Mode:
    1, navigate to the settings page.
    2, Select the “Appearance” tab.
    3, Under the “Theme” section, click the “Dark” option.
